be/src/exec/vectorized/sorting/sort_column.cpp

```diff
@@ -453,7 +453,14 @@ Status sort_vertical_chunks(const std::atomic<bool>& cancel, const std::vector<C
     }
 
     if (limit < perm.size()) {
-        perm.resize(limit);
+        if (is_limit_by_rank) {
+            int first_non_equal_pos = SIMD::find_zero(tie, limit);
```

Comment:
Why the zero in tie indicates rank? It's not so intuitive.

Reply:
Here is the example

Given the number array `[1, 1, 1, 1, 1, 1, 1, 2, 2, 3, 4, 5]`, and the corresponding rank array is `[1, 1, 1, 1, 1, 1, 1, 8, 8, 10, 11, 12]`, the tie array is `[1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1]`

Suppose we need to find rank-5, the 5-th of number array is `1`, we need to consider the trailing equal values, but how to find the end of the trailing equals range? Use the tie structure, we can find the first non-equal value by `SIMD::find_zero(tie, limit)`